### ISSUE TITLE
Can't get shader examples to run via emscripten

### ISSUE THREAD

**Issue by connorjclark** [liballeg/allegro5#1315]
I wanted to see `ex_palette` running in the browser, but I get this error after building locally and opening in the browser:

```
Uncaught TypeError: Failed to execute 'attachShader' on 'WebGLRenderingContext': parameter 2 is not of type 'WebGLShader'
```

I noticed all the shader examples are not presented in the web demo page: https://liballeg.org/examples_graphics.html

I understand that some features of OpenGL are not able to be emulated through WebGL, but as long as you are mindful of the differences in feature sets shouldn't it be possible to build (with emscripten) an allegro5 app that uses shaders?

EDIT

I overlooked a few examples that indeed are working on the demo page, such as https://allegro5.org/examples/examples/ex_shader.html . I guess there's something about `ex_palette` that fails under webgl.

**Comment by connorjclark** [comment]
I build with `-g3` and got a more useful stack trace:

<img width="976" alt="image" src="https://user-images.githubusercontent.com/4071474/156919047-871b59ae-24f3-464b-b329-e1701d235858.png">

**Comment by connorjclark** [comment]
I tracked it down to a compile error for one of the shader programs. I see the error is logged with `ALLEGRO_ERROR` but unfortunately that never shows up in the console. I was able to view the error via chrome devtools stack inspector, though it isn't too straightforward haha:

<img width="995" alt="image" src="https://user-images.githubusercontent.com/4071474/156919852-79a4a853-79e5-4328-b963-4bbf4dabea37.png">

The fragment shaders needs (at least) two changes:

- `precision mediump float;` at the top (webgl requires a precision and does not give a default for fragment shaders)
- `Index expression must be constant` thats the error after fixing the first error. The fragment shader uses a uniform to assign the palette data, which requires accessing via a non-constant expression. Instead, the shader should use a texture to store the palette data.

I assume a portable example would be preferred, so I'll work on submitting a PR that updates this demo.
